enterprise/server/workflow/service/service.go

```diff
@@ -153,19 +154,26 @@ type startWorkflowTask struct {
 	workflow    *tables.Workflow
 }
 
+type StepMaker func(workflow *tables.Workflow) []*rnpb.Step
+
 type workflowService struct {
 	env environment.Env
 
-	wg    sync.WaitGroup
-	tasks chan *startWorkflowTask
-	bbUrl *url.URL
+	wg           sync.WaitGroup
+	tasks        chan *startWorkflowTask
+	bbUrl        *url.URL
+	builtInSteps map[string]StepMaker // Actions that are built-in to the workflow service.
 }
 
 func NewWorkflowService(env environment.Env) *workflowService {
 	ws := &workflowService{
 		env:   env,
 		tasks: make(chan *startWorkflowTask, webhookWorkerTaskQueueSize),
 		bbUrl: build_buddy_url.WithPath(""),
+		builtInSteps: map[string]StepMaker{
+			"@kythe":            config.KytheIndexingAction,
+			"@codesearch_index": config.CodesearchIncrementalUpdateAction,
+		},
```

Comment:
I like this - in the future I hope `uses` can support custom reusable actions, so it might be good to format this as a repo name from the start, like `buildbuddy-io/actions/kythe-index@latest`, `buildbuddy-io/actions/codesearch-index@latest`. Then in the future, we can transparently swap out the implementation so that it fetches from the repo `github.com/buildbuddy-io/actions` and looks for `./codesearch-index/manifest.yaml` or something like that (can be done in a single API request, similar to how we fetch `buildbuddy.yaml`)